tests/stencils/test_fields.py

```diff
@@ -0,0 +1,83 @@
+from dusk.script import *
+from test_util import transpile_and_validate
+
+
+def test_field():
+    transpile_and_validate(temp_field)
+    transpile_and_validate(temp_field_demoted)
+    transpile_and_validate(hv_field)
+    transpile_and_validate(h_offsets)
+    transpile_and_validate(v_offsets)
+    transpile_and_validate(hv_offsets)
+
+
+@stencil
+def temp_field(a: Field[Edge, K], b: Field[Edge, K], out: Field[Edge, K]):
+    x: Field[Edge]
+    with levels_downward as k:
+        x = 1  # stricly necessary in dawn
+        if a > 5:
+            x = a
+        else:
+            x = b
+    with levels_downward as k:
+        out = x
+
+
+@stencil
+def temp_field_demoted(a: Field[Edge, K], b: Field[Edge, K], out: Field[Edge, K]):
+    x: Field[Edge]
+    y: Field[Edge, K]
+    with levels_downward:
+        x = a + b
+        y = 5
+        if x > 3:
+            out = x
+        else:
+            out = y
+
+
+@stencil
+def hv_field(
+    out: Field[Edge, K],
+    full: Field[Edge, K],
+    horizontal: Field[Edge],
+    horizontal_sparse: Field[Edge > Cell],
+    vertical: Field[K],
+):
+    with levels_downward as k:
+        out = full + horizontal + vertical
+        out = reduce_over(Edge > Cell, horizontal_sparse, sum)
+        out = sum_over(Edge > Cell, horizontal_sparse)
+
+
+@stencil
+def h_offsets(
+    a: Field[Edge > Cell > Edge, K], b: Field[Edge, K], c: Field[Edge > Cell > Edge, K]
+):
+    with levels_upward:
+        with sparse[Edge > Cell > Edge]:
+            a = b[Edge] + c  # no offsets, defaults to True
```

Comment:
```suggestion
            a = b[Edge] + c  # no offsets, defaults to Edge > Cell > Edge
```

Reply:
Yes, that's correct. Actually, this is an interesting case, because our current _spec_ isn't very clear on this. Technically, we said it's mandatory to specify for the ambiguous case. But it's not clear what exactly is meant with _ambiguous case_. This could mean:

1) When the start and end location of a _neighborhood iteration_ (reduction & sparse fill) are the same.
2) When the indexing is ambiguous.

E.g.
```python
@stencil
def single_reduction(dense: Field[Edge], sparse: Field[Edge > Cell > Edge]):
  with levels_upward:
    dense = sum_over(
      Edge > Cell > Edge,
      sparse # this is ambiguous for dense, but not sparse
    )

@stencil
def nested_reduction(dense: Field[Edge], sparse: Field[Edge > Cell > Edge]):
  with levels_upward:
    dense = sum_over(
      Edge > Cell > Edge,
      sum_over(
        Edge > Cell > Edge,
        sparse[Edge] # this is ambiguous for both dense and sparse
        # However, if the second chain was different, it wouldn't be
      )
    )
```

But these are very exceptional cases and they get pretty complicated. I think currently 2) is implemented, but I think it's buggy for the nested ambiguous case. Maybe we should track this in a separate issue.

Reply:
I have a note in the back of my TODO to clarify the notation and meaning of neighbor chains in nested reductions in both dawn and dusk. 
Since these ambiguous cases are rare, let's just revisit them in the future or if they cause problems. Agree to track in a separate issue, but let's keep it very low priority.

Reply:
Tracked here #39